lib/viewmodels/map/map_pin_view_model.dart

```diff
@@ -61,6 +65,35 @@ class MapPinViewModel extends AutoDisposeAsyncNotifier<List<MapPinDetails>> {
     return userPosts.map(_toMapPinDetails).toList();
   }
 
+  /// Get user active challenges
+  Future<List<MapPinDetails>> _getUserChallenges() async {
+    final postRepository = ref.watch(postRepositoryServiceProvider);
+    final challengeRepostory = ref.watch(challengeRepositoryServiceProvider);
+    final userId = ref.watch(validLoggedInUserIdProvider);
+    // Only doing a read here, to decrease the number of database reads
+    // (we don't want to re-read the challenges when the position changes).
+    final position = await ref.read(livePositionStreamProvider.future);
+
+    if (position == null) {
+      return List.empty();
+    }
```

Comment:
This is a great idea to only read on the position.

Moreover if we just want the read the position we can use the `geolocationServiceProvider` provider that directly gives us a non-null `GeoPoint`.

```suggestion
    final position =
        await ref.read(geolocationServiceProvider).getCurrentPosition();
```

Reply:
Good catch, I didn't think of that! Great job :)
Fixed in e26a8ca23ae8c739445548da158b4fd4a58c7463.